language/transaction-builder/generator/src/typescript.rs

```diff
@@ -0,0 +1,408 @@
+                     .join(", ")
+        )?;
+        writeln!(self.out, "  ]")?;
+        writeln!(self.out, "}},")?;
+        Ok(())
+    }
+
+    fn output_script_args_callbacks(&mut self, abi: &ScriptABI) -> Result<()> {
+        let mut args_with_types = abi.ty_args()
+            .iter()
+            .map(|ty_arg| format!("{}: DiemTypes.TypeTagVariantStruct", ty_arg.name().to_mixed_case()))
+            .collect::<Vec<_>>();
+        args_with_types.extend(abi.args()
+            .iter()
+            .map(|arg| format!("{}: {}", arg.name().to_mixed_case(), Self::quote_transaction_argument_type(arg.type_tag())))
+            .collect::<Vec<_>>());
+        writeln!(self.out, "{}: (type: string, {}) => void;", abi.name().to_camel_case(), args_with_types.join(", "))?;
+        Ok(())
+    }
+
+    fn quote_doc(doc: &str) -> String {
+        let doc = crate::common::prepare_doc_string(doc);
+        let text = textwrap::indent(&doc, " * ").replace("\n\n", "\n *\n");
+        format!("/**\n{} */\n", text)
+    }
+
+    fn quote_type_parameters(ty_args: &[TypeArgumentABI]) -> Vec<String> {
+        ty_args
+            .iter()
+            .map(|ty_arg| format!("{}: DiemTypes.TypeTag", ty_arg.name()))
+            .collect()
+    }
+
+    fn quote_parameters(args: &[ArgumentABI]) -> Vec<String> {
+        args.iter()
+            .map(|arg| format!("{}: {}", arg.name(), Self::quote_type(arg.type_tag())))
+            .collect()
+    }
+
+    fn quote_type_arguments(ty_args: &[TypeArgumentABI]) -> String {
+        ty_args.iter()
+            .map(|ty_arg| ty_arg.name().to_string())
+            .collect::<Vec<_>>()
+            .join(", ")
+    }
+
+    fn quote_arguments(args: &[ArgumentABI]) -> String {
+        args.iter()
+            .map(|arg| Self::quote_transaction_argument(arg.type_tag(), arg.name()))
+            .collect::<Vec<_>>()
+            .join(", ")
+    }
+
+    fn quote_type(type_tag: &TypeTag) -> String {
+        use TypeTag::*;
+        match type_tag {
+            Bool => "boolean".into(),
+            U8 => "number".into(),
+            U64 => "BigInt".into(),
+            U128 => "BigInt".into(),
+            Address => "DiemTypes.AccountAddress".into(),
+            Vector(type_tag) => match type_tag.as_ref() {
+                U8 => "Uint8Array".into(),
+                _ => common::type_not_allowed(type_tag),
+            },
+
+            Struct(_) | Signer => common::type_not_allowed(type_tag),
+        }
+    }
+
+    fn quote_transaction_argument_type(type_tag: &TypeTag) -> String {
+        use TypeTag::*;
+        match type_tag {
+            Bool => format!("DiemTypes.TransactionArgumentVariantBool"),
+            U8 => format!("DiemTypes.TransactionArgumentVariantU8"),
+            U64 => format!("DiemTypes.TransactionArgumentVariantU64"),
+            U128 => format!("DiemTypes.TransactionArgumentVariantU128"),
+            Address => format!("DiemTypes.TransactionArgumentVariantAddress"),
```

Comment:
No reason to use `format` here. Can use basic string functions (e.g., `to_string()`, maybe `String::from()`).

That is why lint is complaining.

Reply:
You're right. It's leftover from a previous version. 
Will fix it asap.

Reply:
```suggestion
            Bool => "DiemTypes.TransactionArgumentVariantBool".to_string(),
            U8 => "DiemTypes.TransactionArgumentVariantU8".to_string(),
            U64 => "DiemTypes.TransactionArgumentVariantU64".to_string(),
            U128 => "DiemTypes.TransactionArgumentVariantU128".to_string(),
            Address => "DiemTypes.TransactionArgumentVariantAddress".to_string(),
```